src/riak_kv_wm_timeseries.erl

```diff
@@ -29,572 +34,375 @@
+post_is_create(RD, Ctx) ->
+    {false, RD, Ctx}.
 
-validate_request_v1(RD, Ctx = #ctx{method = 'DELETE', table = Table,
-                                  mod = Mod, ddl = DDL}) ->
-    KeysInUrl = lists:map(fun mochiweb_util:unquote/1, wrq:path_tokens(RD)),
-    case path_elements_to_key(Table, KeysInUrl, Mod, DDL) of
-        {ok, Key} ->
-            valid_params(
-              RD, Ctx#ctx{api_version = "v1", api_call = delete,
-                          key = Key});
-        {error, Reason} ->
-            handle_error(Reason, RD, Ctx)
+-spec process_post(#wm_reqdata{}, #ctx{}) ->  cb_rv_spec(boolean()).
+process_post(RD, #ctx{mod=Mod,
+                     table=Table}=Ctx) ->
+    try extract_data(RD, Mod) of
+        Records ->
+            case riak_kv_ts_util:validate_rows(Mod, Records) of
+                [] ->
+                    case riak_kv_ts_api:put_data(Records, Table, Mod) of
+                        ok ->
+                            Json = result_to_json(ok),
+                            Resp = riak_kv_wm_ts_util:set_json_response(Json, RD),
+                            {true, Resp, Ctx};
+                        {error, {some_failed, ErrorCount}} ->
+                            Resp = riak_kv_wm_ts_util:set_error_message("failed some puts ~p ~p",
+                                                                        [ErrorCount, Table],
+                                                                        RD),
+                            {{halt, 400}, Resp, Ctx}
+                    end;
+                BadRowIdxs when is_list(BadRowIdxs) ->
+                    Resp = riak_kv_wm_ts_util:set_error_message("invalid data: ~p",
+                                                                [BadRowIdxs],
+                                                                RD),
+                    {{halt, 400}, Resp, Ctx}
+            end
+    catch
+        throw:{data_problem,Reason} ->
+            Resp = riak_kv_wm_ts_util:set_error_message("wrong body: ~p", [Reason], RD),
+            {{halt, 400}, Resp, Ctx}
     end.
 
-
--spec extract_data([byte()]) -> undefined|any().
-extract_data(Json) ->
-    try mochijson2:decode(Json) of
-        Decoded when is_list(Decoded) ->
-            validate_ts_records(Decoded)
+-spec delete_resource(#wm_reqdata{}, #ctx{}) ->  cb_rv_spec(boolean()|halt()).
+delete_resource(RD,  #ctx{table=Table,
+                          mod=Mod,
+                          key=Key,
+                          options=Options}=Ctx) ->
+     try riak_kv_ts_api:delete_data(Key, Table, Mod, Options) of
+        ok ->
+             Json = result_to_json(ok),
+              Resp = riak_kv_wm_ts_util:set_json_response(Json, RD),
+              {true, Resp, Ctx};
+         {error, notfound} ->
+              {{halt, 404}, RD, Ctx}
+     catch
+         _:Reason ->
+             lager:log(info, self(), "delete_resource failed: ~p", [Reason]),
```

Comment:
The lager things should have been removed before doing the PR - I will remove them all before asking for a re-review.